src/duration.rs

```diff
@@ -436,13 +436,86 @@ impl Duration {
     }
 
     fn parse_time(bytes: &[u8], offset: usize) -> Result<Self, ParseError> {
-        let t = Time::parse_bytes_offset(bytes, offset)?;
+        if bytes.len() - offset < 5 {
```

Comment:
Looks like this duplicates the logic from `Time::parse_bytes_offset` I think we should have a new method on `TIme`, with signature something like

```rs
pub(crate) fn parse_hms(bytes: &[u8], offset: usize) -> Result<(u8, u8, u8), ParseError> {
```

Which we use both here and in `time.rs`.

Reply:
After taking a look, I guess this is not possible because the `length` variable after calculating the `HH:mm:SS...` is needed in `time.rs` in order to get the TzInfo. Otherwise it cannot raise errors for extra characters explicitly.

- We can only have `parse_hm(..)` and parsing second and microsecond should be inlined in both files
- Or change signature to something like `pub(crate) fn parse_partial_time(bytes: &[u8], offset: usize) -> Result<(u8, u8, u8, u32, u8), ParseError> {` to return `HH:mm:SS::fffff` and the length of the parsed bytes to know if optional seconds and microseconds were parsed.